drivers/crypto/crypto_it8xxx2_sha.c

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) 2023 ITE Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ite_it8xxx2_sha
+
+#include <zephyr/kernel.h>
+#include <zephyr/crypto/crypto.h>
+#include <zephyr/sys/byteorder.h>
+#include <chip_chipregs.h>
+#include <errno.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(sha_it8xxx2, CONFIG_CRYPTO_LOG_LEVEL);
+
+#define IT8XXX2_SHA_REGS_BASE \
+	((struct it8xxx2_sha_regs *)DT_REG_ADDR(DT_NODELABEL(sha0)))
+
+#define IT8XXX2_SHA_START_SHA256 BIT(1)
+
+struct chip_sha256_ctx {
+	union {
+		/* W[0] ~ W[15] */
+		uint32_t w_sha[16];
+		uint8_t w_input[64];
+	};
+	/* reserved */
+	uint32_t reserved1[8];
+	/* H[0] ~ H[7] */
+	uint32_t h[8];
+	/* reserved */
+	uint32_t reserved2[30];
+	uint32_t w_input_index;
+	uint32_t total_len;
+	/* K[0] ~ K[63] */
+	uint32_t k[64];
```

Comment:
Please declare constants for the array lengths used in this structure.  You should use the same constants for the `sha256_h0[]` and `sha256k_k[]` arrays.

Reply:
Done